src/views/dashboard/ProjectForm.tsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import Button from "@material-ui/core/Button";
+import { PROJECT } from "../../graphql-requests/mutations";
+import { useMutation } from "@apollo/react-hooks";
+
+type FormData = {
+   project_name: String;
+   project_developer: String;
+   project_description: String;
+   completed: Boolean;
+};
+
+export const Project = (props: any) => {
+   const { register, handleSubmit } = useForm<FormData>();
+   const [project] = useMutation(PROJECT);
+   const onSubmit = ({
+      project_name,
+      project_developer,
+      project_description,
+      completed,
+   }: FormData) => {
+      project({
+         variables: {
+            project_name: project_name,
+            project_developer: project_developer,
+            project_description: project_description,
+            completed: completed,
+         },
+      })
+         .then(res => {
+            //console.log(res.data);
+            localStorage.setItem("project_name", res.data.project.project_name);
+         })
+         .then(data => {
+            props.history.push("/home");
+            console.log("Registered Project");
+         })
+         .catch(err => err.message);
+   };
+
+   return (
+      <>
+         <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
We'll need to refactor the form to utilize the new Form and Input components.

Reply:
Agreed, my next todo